segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -212,6 +219,25 @@ public void setupAppend(SetupAppend setupAppend) {
                         });
     }
 
+    private void populateEventSizeForIndexSegmentAppends(UUID writer, String indexSegment, long requestId) {
+        store.getStreamSegmentInfo(indexSegment, TIMEOUT)
+                .whenComplete((properties, u) -> {
+                    if (u != null) {
+                        u = Exceptions.unwrap(u);
+                        if (u instanceof NoSuchSegmentException || u instanceof StreamSegmentNotExistsException) {
+                            log.error("could not find the segment {}", indexSegment);
+                        } else {
+                            handleException(writer, requestId, indexSegment, "populating event size for index appends", u);
+                        }
+                    } else {
+                        if (properties != null) {
+                            Map<AttributeId, Long> attributes =  properties.getAttributes();
+                            this.writerStates.put(Pair.of(indexSegment, writer), new WriterState(attributes.get(Attributes.ALLOWED_INDEX_SEG_EVENT_SIZE)));
```

Comment:
Do we need to initiate event number for index segment?

Reply:
No, that is being handled in the append part.

Reply:
Need of writerStates map if segement event size is constant is not very clear to me

Reply:
This is not being used currently, but this code will be used once we get clarity on how and what will be the value for this attribute which we are storing in the cache.